src/PHPCR/Util/QOM/Sql2ToQomQueryConverter.php

```diff
@@ -728,52 +736,54 @@ protected function parsePropertyValue()
 
     protected function parseCastLiteral($token)
     {
-        if ($this->scanner->tokenIs($token, 'CAST')) {
-            $this->scanner->expectToken('(');
-            $token = $this->scanner->fetchNextToken();
+        if (!$this->scanner->tokenIs($token, 'CAST')) {
+            throw new \LogicException('parseCastLiteral when not a CAST');
+        }
```

Comment:
git diff really mangles this one. all i did is change the ``if ($this->scanner->tokenIs()...`` to a `if (!$this->scanner->tokenIs()... throw \LogicException` and then unindented all the code below without changing anything.